android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/GpsFactory.java

```diff
@@ -224,27 +223,40 @@ protected void addViewTags(Button recordButton, WidgetArgs widgetArgs, WidgetMet
         recordButton.setTag(R.id.extraPopup, widgetArgs.isPopup());
     }
 
-    public void attachLayout(Context context, @NonNull JSONObject jsonObject, @NonNull View dataView, @NonNull TextView latitudeTv, @NonNull TextView longitudeTv, @NonNull TextView altitudeTv, @NonNull TextView accuracyTv) {
-        String latitude = "";
-        String longitude = "";
-        String accuracy = "";
-        String altitude = "";
+    public void attachLayout(Context context, @NonNull JSONObject jsonObject, @NonNull View dataView,
+                             @NonNull TextView latitudeTv, @NonNull TextView longitudeTv,
+                             @NonNull TextView altitudeTv, @NonNull TextView accuracyTv) {
+        TextView[] views = new TextView[]{latitudeTv, longitudeTv, altitudeTv, accuracyTv};
+
+        int[] stringIds = new int[]{R.string.latitude, R.string.longitude, R.string.altitude, R.string.accuracy};
+
+        //loops to fill all textViews with empty string
+        for (int i = 0; i < stringIds.length; i++) {
+            views[i].setText(getText(context, "", stringIds[i]));
+        }
+
         if (jsonObject.has(JsonFormConstants.VALUE)) {
             String coordinateData = jsonObject.optString(JsonFormConstants.VALUE);
 
             String[] coordinateElements = coordinateData.split(" ");
-            if (coordinateElements.length > 1) {
-                latitude = coordinateElements[0];
-                longitude = coordinateElements[1];
+
+            //loops to fill textViews with values supplied
+            for (int i = 0; i < coordinateElements.length; i++) {
+                views[i].setText(getText(context, coordinateElements[i], stringIds[i]));
             }
-        }
 
-        latitudeTv.setText(getText(context, latitude, R.string.latitude));
-        longitudeTv.setText(getText(context, longitude, R.string.longitude));
-        altitudeTv.setText(getText(context, altitude, R.string.altitude));
-        accuracyTv.setText(getText(context, accuracy, R.string.accuracy));
+            dataView.setTag(R.id.raw_value, constructString(coordinateElements));
+        } else {
+            dataView.setTag(R.id.raw_value, "");
```

Comment:
No need for this else block, we can set `dataView.setTag(R.id.raw_value, "");` as the default value to be overwritten only if we have gps values.